test/e2e_node/services/apiserver.go

```diff
@@ -96,9 +102,14 @@ func (a *APIServer) Name() string {
 }
 
 func getAPIServerClientURL() string {
-	return apiserverClientURL
+	return framework.TestContext.Host
```

Comment:
What does this get set to? Because you're hardcoding the bind address to `127.0.0.1`, so I think anything else (except maybe localhost) won't work. Note that the bind address MUST be 127.0.0.1 as long as you're using the `insecure-token`.

Reply:
Oh I forgot the listen address is hard-coded.. then do we need to enforce `framework.TestContext.Host` to be `https://127.0.0.1:6443` as well?

Reply:
Maybe we should just default the TestContext.Host to localhost? (only for the node suite though)

Reply:
`TestContext.Host` is actually the `defaultHost` (i.e. `https://127.0.0.1:6443`)https://github.com/kubernetes/kubernetes/blob/7a2812ca6087e62d0c9b6dde050065fe2f665a0e/test/e2e/framework/test_context.go#L419-L421 in node e2e test.